packages/tools/src/tools/SculptorTool.ts

```diff
@@ -0,0 +1,590 @@
+  viewportIdsToRender: any[];
+  isEditingOpenContour: boolean;
+  canvasLocation: Types.Point2 | undefined;
+};
+
+class SculptorTool extends BaseTool {
+  static toolName: string;
+  registeredShapes = new Map();
+  private isActive = false;
+  private selectedShape: string;
+  private commonData: CommonData = {
+    activeAnnotationUID: null,
+    viewportIdsToRender: [],
+    isEditingOpenContour: false,
+    canvasLocation: undefined,
+  };
+  private sculptData?: SculptData;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        minSpacing: 1,
+        referencedToolNames: [
+          'PlanarFreehandROI',
+          'PlanarFreehandContourSegmentationTool',
+        ],
+        toolShape: 'circle',
+        referencedToolName: 'PlanarFreehandROI',
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+    this.endCallback = this.endCallback.bind(this);
+    this.dragCallback = this.dragCallback.bind(this);
+    this.registerShapes(CircleSculptCursor.shapeName, CircleSculptCursor);
+    this.setToolShape(this.configuration.toolShape);
+  }
+
+  /**
+   * Register different tool shapes for sculptor tool
+   * @param shapeName name of shape
+   * @param shapeClass shape class
+   */
+  registerShapes(shapeName: string, shapeClass: any): void {
+    const shape = new shapeClass();
+    this.registeredShapes.set(shapeName, shape);
+  }
+
+  preMouseDownCallback = (evt: EventTypes.InteractionEventType): boolean => {
+    const eventData = evt.detail;
+    const element = eventData.element;
+
+    this.configureToolSize(evt);
+    this.selectFreehandTool(eventData);
+
+    if (this.commonData.activeAnnotationUID === null) {
+      return;
+    }
+
+    this.isActive = true;
+
+    hideElementCursor(element);
+    this.activateSculpt(element);
+    return true;
+  };
+
+  mouseMoveCallback = (evt: EventTypes.InteractionEventType): void => {
+    if (this.mode === ToolModes.Active) {
+      this.configureToolSize(evt);
+      this.updateCursor(evt);
+    } else {
+      this.commonData.canvasLocation = undefined;
+    }
+  };
+
+  /**
+   * Sculpts the freehand ROI with freehandSculpter tool, moving,
+   * adding and removing handles as necessary.
+   *
+   * @param eventData - Data object associated with the event.
+   * @param points - Array of points
+   */
+  protected sculpt(eventData: any, points: Array<Types.Point3>): void {
+    const config = this.configuration;
+    const element = eventData.element;
+
+    const enabledElement = getEnabledElement(element);
+    const { viewport } = enabledElement;
+    const cursorShape = this.registeredShapes.get(this.selectedShape);
+
+    this.sculptData = {
+      mousePoint: eventData.currentPoints.world,
+      mouseCanvasPoint: eventData.currentPoints.canvas,
+      points,
+      maxSpacing: cursorShape.getMaxSpacing(config.minSpacing),
```

Comment:
You should be able to declare an inline interface that has getMaxSpacing and has pushHandles on it